codegen/generator/functions.go

```diff
@@ -102,12 +103,21 @@ func (c *CommonFunctions) GetArrayField(f *introspection.Field) []*introspection
 	}
 
 	var fields []*introspection.Field
+	var idField *introspection.Field
 	// Only include scalar fields for now
 	// TODO: include subtype too
 	for _, typeField := range schemaType.Fields {
 		if typeField.TypeRef.IsScalar() {
 			fields = append(fields, typeField)
 		}
+		// TODO: hack to fix requesting all fields from list of id-able objects, need better solution
```

Comment:
reminder to self: double check this (I think it may actually be totally fine) and spin out to separate PR against main if so